bank_lib/src/bank.rs

```diff
@@ -0,0 +1,315 @@
+        });
+        Ok(op_index)
+    }
+
+    /// Возвращает баланс указанного счета.
+    ///
+    /// # Пример
+    ///
+    /// ```
+    /// use bank_lib::Bank;
+    /// let mut bank = Bank::new();
+    /// bank.create_account("Alice".to_string()).unwrap();
+    /// assert_eq!(bank.get_balance("Alice"), Ok(0));
+    /// ```
+    ///
+    /// # Ошибки
+    ///
+    /// Возвращает `BankError::AccountNotFound`, если счет не существует.
+
+    pub fn get_balance(&self, name: &str) -> Result<i64, BankError> {
+        self.accounts
+            .get(name)
+            .cloned()
+            .ok_or(BankError::AccountNotFound)
+    }
+
+    /// Возвращает операцию по индексу.
+    ///
+    /// # Пример
+    ///
+    /// ```
+    /// use bank_lib::{Bank, Operation};
+    /// let mut bank = Bank::new();
+    /// let op_index = bank.create_account("Alice".to_string()).unwrap();
+    /// assert_eq!(bank.get_operation(op_index).unwrap(), &Operation::CreateAccount { name: "Alice".to_string() });
+    /// ```
+
+    pub fn get_operation(&self, index: usize) -> Option<&Operation> {
+        self.history.get(index)
+    }
+
+    /// Возвращает историю операций для указанного счета.
+    ///
+    /// # Пример
+    ///
+    /// ```
+    /// use bank_lib::Bank;
+    /// let mut bank = Bank::new();
+    /// bank.create_account("Alice".to_string()).unwrap();
+    /// bank.deposit("Alice", 100).unwrap();
+    /// let history = bank.get_account_history("Alice").unwrap();
+    /// assert_eq!(history.len(), 2);
+    /// ```
+    ///
+    /// # Ошибки
+    ///
+    /// Возвращает `BankError::AccountNotFound`, если счет не существует.
+
+    pub fn get_account_history(&self, name: &str) -> Result<Vec<&Operation>, BankError> {
+        let indices = self
+            .account_operations
+            .get(name)
+            .ok_or(BankError::AccountNotFound)?;
+        let operations = indices.iter().map(|&i| &self.history[i]).collect();
+        Ok(operations)
+    }
+
+    /// Возвращает всю историю операций.
+    ///
+    /// # Пример
+    ///
+    /// ```
+    /// use bank_lib::Bank;
+    /// let mut bank = Bank::new();
+    /// bank.create_account("Alice".to_string()).unwrap();
+    /// bank.deposit("Alice", 100).unwrap();
+    /// let history = bank.get_all_history();
+    /// assert_eq!(history.len(), 2);
+    /// ```
+
+    pub fn get_all_history(&self) -> &[Operation] {
+        &self.history
+    }
+
+    /// Восстанавливает состояние банка, повторно выполняя все операции из истории.
+    ///
+    /// # Пример
+    ///
+    /// ```
+    /// use bank_lib::Bank;
+    /// let mut bank = Bank::new();
+    /// bank.create_account("Alice".to_string()).unwrap();
+    /// bank.deposit("Alice", 100).unwrap();
+    /// let history = bank.get_all_history().to_vec();
+    /// let new_bank = Bank::replay_operations(&history);
+    /// assert_eq!(new_bank.get_balance("Alice"), Ok(100));
+    /// ```
+    pub fn replay_operations(operations: &[Operation]) -> Self {
```

Comment:
Учитывая мой коммент выше про итератор по истории аккаунтов, то предлагаю сделать так:

```suggestion
    pub fn replay_operations(operations: impl Iterator<Item = &Operation>) -> Self {
```

Ну а вдруг кто-нибудь захочет сделать реплей из операций только одного аккаунта?